### PR TITLE
fix: publish diagnostics from cache when file is opened

### PR DESCRIPTION
**Pull request by bastiandoetsch**
### Description

fix: report diagnostics from cache on "textDocument/didOpen"

Fixes switching between files in sublime, as sublime expects diagnostics to be reported anew when a file is opened.

Issue filed under: https://github.com/sublimelsp/LSP/issues/2200

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [ ] README.md updated, if user-facing
- [ ] License file updated, if new 3rd-party dependency is introduced

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: application/server/server.go [hunk 1]
```diff
@@ -367,6 +372,26 @@ func logError(err error, method string) {
...
 }
 
+func textDocumentDidOpenHandler() jrpc2.Handler {
+	return handler.New(func(_ context.Context, params sglsp.DidOpenTextDocumentParams) (any, error) {
+		filePath := uri.PathFromUri(params.TextDocument.URI)
+		logger := log.With().Str("method", "TextDocumentDidOpenHandler").Str("documentURI", filePath).Logger()
+
+		logger.Info().Msg("Receiving")
+		folder := workspace.Get().GetFolderContaining(filePath)
+		issues := folder.DocumentDiagnosticsFromCache(filePath)
```
Comment (michelkaporin) [inline application/server/server.go:382]: This doesn't respect severity filtering.

**Review by michelkaporin — CHANGES_REQUESTED** [review]

File: application/server/server.go (hunk 1 above)
Comment (bastiandoetsch) [reply application/server/server.go:382]: Ah yeah, good call.

Comment (bastiandoetsch) [reply application/server/server.go:382]: oops. not fixed yet.

File: application/server/server.go [hunk 2]
```diff
@@ -367,6 +372,27 @@ func logError(err error, method string) {
...
+		if len(issues) > 0 {
+			logger.Info().Msg("Sending cached issues")
+			filteredIssues := workspace.FilterIssues(issues, config.CurrentConfig().GetDisplayableIssueTypes())
```
Comment (bastiandoetsch) [inline application/server/server.go:385]: make sure filtered issues is checked.

**Review by PeterSchafer — APPROVED** [review]
LGTM!

**Review by michelkaporin — APPROVED** [review]
👌

**Pull request merged**